modules/Microsoft.CognitiveServices/accounts/deploy.bicep

```diff
@@ -215,11 +242,20 @@ resource cognitiveServices 'Microsoft.CognitiveServices/accounts@2021-10-01' = {
     allowedFqdnList: allowedFqdnList
     apiProperties: apiProperties
     disableLocalAuth: disableLocalAuth
-    encryption: !empty(encryption) ? encryption : null
+    encryption: !empty(cMKKeyName) ? {
```

Comment:
If  `cMKKeyName` makes all other `keyVaultProperties` required, then I guess all other `keyVaultProperties` should also have a `Conditional` parameter category.

Reply:
Updated whereever applicable

Reply:
Although out of the scope of this PR, but since we started, I think we need to align the implementation and parameter category of `cMKUserAssignedIdentityResourceId` as well where applicable. 

ML workspaces and ServiceBus use `cMKUserAssignedIdentityResourceId`, but implemented like 
```
identity: !empty(cMKUserAssignedIdentityResourceId) ? {
        userAssignedIdentity: cMKUserAssignedIdentityResourceId
      } : null
```
which correctly makes the parameter `Optional`. 
- If the same implementation makes sense also for automation account and cognitive services, then I'd suggest to align that, updating the identity implementation and changing `cMKUserAssignedIdentityResourceId` from `Conditional` to `Optional`
- otherwise, we should update the `cMKUserAssignedIdentityResourceId` category from `Optional` to `Conditional` in the automation account module, since its current implementation depends on `cMKKeyName`.

Reply:
It is in fact different from resource to resource. Some enforce User-Assigned-Identities, some allow System-Assigned-Identities, etc.

I would highly recommend to keep this out of this PR. If we don't trust the implementations that are there now, we should revist all components and test again the difference scenarios.

Reply:
I do agree however, that it should match the current implementation. So in your example above we should use 'Optional' - not 'Conditional'

Reply:
Here it seems to be required

Reply:
addressed